cmd/generate/Dockerfile.template

```diff
@@ -1,20 +1,34 @@
 # DO NOT EDIT! Generated Dockerfile for {{.main}}.
-FROM {{.builder}} as builder
+ARG GO_BUILDER={{.builder}}
+ARG GO_RUNTIME=registry.access.redhat.com/ubi8/ubi-minimal
+
+FROM $GO_BUILDER as builder
 
 COPY . .
 
-RUN mkdir -p /var/run/ko && \
-    mkdir -p {{.main}}/kodata && \
-    go build -o /usr/bin/main ./{{.main}} && \
-    cp -r {{ .main }}/kodata /var/run/ko
+ENV CGO_ENABLED=1
+ENV GOEXPERIMENT=strictfipsruntime
+
+RUN go build -tags strictfipsruntime -o /usr/bin/main ./{{.main}}
+
+FROM $GO_RUNTIME
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal
+ARG VERSION={{.version}}
```

Comment:
This version needs to include the patch version as well for the release, e.g. `1.33.2`. 
It's not clear to me how we would pass the exact version.
An idea:
* For the given upstream branch, [guess ](https://github.com/openshift-knative/hack/blob/main/pkg/sobranch/version.go#L10) the serverless-operator branch
* Checkout the repo/branch of serverless-operator 
* Read project.yaml from s-o (key `project.version`)
* Use the value and generate Tekton pipelines for konflux that would pass the version as build-args to Buildah

@pierDipi WDYT?

Reply:
That logic sounds good to me, one question, is that `version` label required ? or are we doing that mostly for compatibility with previous ones?

Reply:
> Use the value and generate Tekton pipelines for konflux that would pass the version as build-args to Buildah

We have only pipelines per branch. Or do you mean having a separate step/task in the pipeline, which gets the exact version (including patch version) and pass this to the buildah task?

Reply:
I was thinking that pipelines-per-branch are good enough. When the Tekton Pipelines are generated for the given branch, we can already set the exact version based on project.yaml (i.e. 1.34.0). Any Konflux builds will have this exact version in the LABEL until the version is again bumped for another z-stream release (e.g. 1.34.1). From that moment any further builds will again have 1.34.1 in the LABEL. Is that a problem?

Reply:
+1 on doing that, all git driven as opposed to magic pipeline steps

Reply:
> is that version label required ?
I think the information is used for the Red Hat image catalog:

![image](https://github.com/user-attachments/assets/e80d06f0-222c-4a69-a436-310ab688f1ea)

Reply:
> From that moment any further builds will again have 1.34.1

Where would you get this information from in your pipeline?

Reply:
Yes it is mandatory labels for OSBS/brew build, I'm not sure about konflux use the same way
and these label are also use display at redhat catalog page.

Reply:
> Where would you get this information from in your pipeline? (I mean that you're now on 1.34.1 instead of 1.34.0)

From project.yaml in serverless-operator repo, as mentioned in the first comment of this thread.

Reply:
> From project.yaml in serverless-operator repo, as mentioned in the first comment of this thread.

yes, but it is unclear for me, how to do this in plain yaml and pass it to the buildah task?

Reply:
Or would you regenerate the pipeline yamls each time we have a new patch release created?

Reply:
> Or would you regenerate the pipeline yamls each time we have a new patch release created?

This, similar to https://github.com/openshift-knative/serverless-operator/pull/2823 when pushing to `release-*` branches

Reply:
not the pipeline YAML though, the dockerfiles

Reply:
Ok :+1: 
But I am still not sure, if having a dedicated task to get the version from the project.yaml wouldn't be less disruptive as updating the yaml files each time. But I don't have a strong opinion on it...

Reply:
Pushed the commit below that implements it.